src/custom-element.js

```diff
@@ -1210,6 +1210,17 @@ function createBaseCustomElementClass(win) {
     }
 
     /**
+     * Called when an attribute's value changes.
+     * Only called for observedAttributes or from amp-bind.
+     * @param {!string} name
+     * @param {?string} oldValue
+     * @param {?string} newValue
+     */
+    attributeChangedCallback(name, oldValue, newValue) {
```

Comment:
@cramforce Could you elaborate on what you mean by that? Add specific APIs for each AMP extension like `srcChangedCallback` to be called by Bind? Or an API that supports bulk changes like Dima suggested?

Also note that the browser only calls `attributeChangedCallback` for attributes specified in [`observedAttributes`](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Custom_Elements#Observed_attributes), which is currently **not** used anywhere in AMP.